jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessor.java

```diff
@@ -289,16 +273,34 @@ static PluginConfigurationProcessor processCommonConfiguration(
    * @param rawConfiguration raw configuration data
    * @param projectProperties used for providing additional information
    * @return the base image
+   * @throws IncompatibleBaseImageJavaVersionException
    */
   @VisibleForTesting
-  static String getBaseImage(
-      RawConfiguration rawConfiguration, ProjectProperties projectProperties) {
-    return rawConfiguration
-        .getFromImage()
-        .orElse(
-            projectProperties.isWarProject()
-                ? "gcr.io/distroless/java/jetty"
-                : "gcr.io/distroless/java");
+  static String getBaseImage(RawConfiguration rawConfiguration, ProjectProperties projectProperties)
+      throws IncompatibleBaseImageJavaVersionException {
+    int javaVersion = projectProperties.getMajorJavaVersion();
+
+    if (rawConfiguration.getFromImage().isPresent()) {
+      String baseImage = rawConfiguration.getFromImage().get();
+
+      if (knownDistrolessJava8Image(baseImage) && javaVersion > 8) {
+        throw new IncompatibleBaseImageJavaVersionException(8, javaVersion);
+      } else if (knownDistrolessJava11Image(baseImage) && javaVersion > 11) {
+        throw new IncompatibleBaseImageJavaVersionException(11, javaVersion);
+      }
+      return baseImage;
+    }
+
+    // Base image not configured; auto-pick Distroless.
+    if (projectProperties.isWarProject()) {
+      return "gcr.io/distroless/java/jetty";
+    } else if (javaVersion <= 8) {
+      return "gcr.io/distroless/java:8";
+    } else if (javaVersion <= 11) {
+      return "gcr.io/distroless/java:11";
+    } else {
+      throw new IncompatibleBaseImageJavaVersionException(11, javaVersion);
```

Comment:
Should this use `11` as the `baseImageMajorJavaVersion`?

Reply:
We are auto-picking a base image and the highest available now is 11, so I thought saying 11 works here.